src/main/java/by/itech/lab/supplier/service/mail/MailServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package by.itech.lab.supplier.service.mail;
+
+
+import by.itech.lab.supplier.domain.User;
+import by.itech.lab.supplier.dto.UserDto;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.stringtemplate.v4.ST;
+import org.thymeleaf.TemplateEngine;
+
+@Service
+@AllArgsConstructor
+public class MailServiceImpl implements MailService {
+    private static final String LASTNAME = "username";
+    private static final String LINK = "link";
+    private static final String USERNAME = "adminMail";
+    private static final String PASSWORD = "password";
+    private static final String ACTIVATION_LINK = "Activation link";
+
+    public void sendMail(UserDto user) {
+            Sender sender = new Sender(USERNAME, PASSWORD);
+            Email email = new Email();
+            email.setTo(user.getEmail());
+            email.setSubject(ACTIVATION_LINK);
+            email.setText(buildMessage(user, String.format("localhost:8080/users/%s/true", user.getUsername())));
+            sender.send(email);
+    }
+
+    private String buildMessage(UserDto user, String url) {
+        ST st = new ST("Hello, <username>! To activate Your profile, please follow the link below <link>");
```

Comment:
Please dont use external libs for things that already exist in Spring, use TemplateEngine, which renders HTML with Thymeleaf

Reply:
I don't understand how to work with temlate engine here

Reply:
```
Map<String, Object> attributes = new HashMap();
attributes.put("firstName", "Alesya");
attributes.put("lastName", "Glushakova");
attributes.put("title", "Registration email");
Context context = new Context(*locale*, attributes)
templateEngine.process("path_to_html_template_in_resources(excluding_.html)", context);
```
HTML Template:
```
<!DOCTYPE html>
<html xmlns:th="http://www.thymeleaf.org">
  <head>
    <title th:text="${title}"></title>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
  </head>
  <body>
    <p th:text="|Hello ${firstName} ${lastName},|"></p>
  </body>
</html>
```